frodo.js

```diff
@@ -1,17 +1,16 @@
 /**
```

Comment:
SCRIPT-This block of code could benefit from some comments explaining the logic behind it.